test-case/check-audio-equalizer.sh

```diff
@@ -52,95 +52,95 @@ func_test_eq()
         dloge "Equalizer setting failure with $conf"
         return 1
     fi
-    # test with aplay
-    dlogc "aplay -D$dev -f $fmt -c $channel -r $rate /dev/zero -d $duration"
-    aplay -D$dev -f $fmt -c $channel /dev/zero -d $duration
+
+    dlogc "$cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file"
+    $cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file
     if [ $? -ne 0 ]; then
         dloge "Equalizer test failure with $conf"
         return 1
     fi
     sleep 1
 }
 
+# this function performs IIR/FIR filter test
+# param1 must be must be iir or fir
+func_test_filter()
+{
+    local testfilter=$1
+    dlogi "1. Get amixer control id for $testfilter"
+    Filterid=`amixer -D hw:$sofcard controls | grep -i eq${testfilter}| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
+    if [ -z "$Filterid" ]; then
```

Comment:
Can `eq${testfilter}` match multiple `controls` lines?? If yes can you add a comment explaining why the first line is "the best"? If not this should probably error, no?

Reply:
I still don't understand the `head -1` but if someone else can then fine.

Reply:
Even without head -1, it works. but it need to be addressed with `TODO` . I will work on it once this merged.

Reply:
This is my comment on latest code.
> TODO: Need to match alsa control id with the filter in the pipeline,
>       currently the test discards EQ pipelines except first one.